src/Bicep.Core/Syntax/SyntaxRewriteVisitor.cs

```diff
@@ -0,0 +1,534 @@
+        {
+            var hasChanges = Rewrite(syntax.Child, out var child);
+
+            if (!hasChanges)
+            {
+                return syntax;
+            }
+
+            return new IdentifierSyntax(child);
+        }
+        void ISyntaxVisitor.VisitIdentifierSyntax(IdentifierSyntax syntax) => ReplaceCurrent(syntax, ReplaceIdentifierSyntax);
+
+        protected virtual TypeSyntax ReplaceTypeSyntax(TypeSyntax syntax)
+        {
+            var hasChanges = Rewrite(syntax.Identifier, out var identifier);
+
+            if (!hasChanges)
+            {
+                return syntax;
+            }
+
+            return new TypeSyntax(identifier);
+        }
+        void ISyntaxVisitor.VisitTypeSyntax(TypeSyntax syntax) => ReplaceCurrent(syntax, ReplaceTypeSyntax);
+
+        protected virtual BooleanLiteralSyntax ReplaceBooleanLiteralSyntax(BooleanLiteralSyntax syntax)
+        {
+            var hasChanges = Rewrite(syntax.Literal, out var literal);
+
+            if (!hasChanges)
+            {
+                return syntax;
+            }
+
+            // TODO calculate value correctly here
+            return new BooleanLiteralSyntax(literal, syntax.Value);
+        }
+        void ISyntaxVisitor.VisitBooleanLiteralSyntax(BooleanLiteralSyntax syntax) => ReplaceCurrent(syntax, ReplaceBooleanLiteralSyntax);
+
+        protected virtual StringSyntax ReplaceStringSyntax(StringSyntax syntax)
+        {
+            var hasChanges = Rewrite(syntax.StringTokens, out var stringTokens);
+            hasChanges |= Rewrite(syntax.Expressions, out var expressions);
+
+            if (!hasChanges)
+            {
+                return syntax;
+            }
+
+            // TODO calculate segmentValues correctly here
+            return new StringSyntax(stringTokens, expressions, syntax.SegmentValues);
+        }
+        void ISyntaxVisitor.VisitStringSyntax(StringSyntax syntax) => ReplaceCurrent(syntax, ReplaceStringSyntax);
+
+        protected virtual ProgramSyntax ReplaceProgramSyntax(ProgramSyntax syntax)
+        {
+            var hasChanges = Rewrite(syntax.Children, out var children);
+            hasChanges |= Rewrite(syntax.EndOfFile, out var endOfFile);
+
+            if (!hasChanges)
+            {
+                return syntax;
+            }
+
+            // TODO what to do with lexerDiagnostics?
+            return new ProgramSyntax(children, endOfFile, syntax.LexerDiagnostics);
+        }
+        void ISyntaxVisitor.VisitProgramSyntax(ProgramSyntax syntax) => ReplaceCurrent(syntax, ReplaceProgramSyntax);
+
+        protected virtual NumericLiteralSyntax ReplaceNumericLiteralSyntax(NumericLiteralSyntax syntax)
+        {
+            var hasChanges = Rewrite(syntax.Literal, out var literal);
+
+            if (!hasChanges)
+            {
+                return syntax;
+            }
+
+            // TODO calculate value correctly here
```

Comment:
>here [](start = 46, length = 4)

What do you mean?

Reply:
If the token has been modified then value (the numeric value of the token) needs to be updated. I've fixed this now.